forge/db/models/User.js

```diff
@@ -99,16 +99,43 @@ module.exports = {
                 // determine if this user owns any teams
                 // throw an error if we would orphan any teams
                 const teams = await app.db.models.Team.forUser(user)
+                const teamBlockers = []
+                const teamOwnerships = []
                 for (const team of teams) {
                     const owners = await team.Team.getOwners()
                     const isOwner = owners.find((owner) => owner.id === user.id)
 
-                    // if this user is the only owner of this team, throw an error
                     if (isOwner && owners.length <= 1) {
-                        throw new Error('Cannot delete the last owner of a team')
+                        const instanceCount = await team.Team.instanceCount()
+                        const deviceCount = await team.Team.deviceCount()
+                        const members = await team.Team.memberCount()
+
+                        teamOwnerships.push(team)
+
+                        // throw error if the team has other members assigned to it
+                        if (members > 1) {
+                            teamBlockers.push(`Team ${team.Team.name} which is being deleted alongside your account still has users in it.`)
+                        }
```

Comment:
Is it a problem if other non owners are in the team?

@joepavitt if a user wants to leave and delete their account, should we prevent this just because a team they are the sole owner of has members?

also, the wording is a little verbose though i cannot think of a good alternative at this time!

Reply:
I asked the same question here https://github.com/FlowFuse/flowfuse/issues/4292#issuecomment-2275725114

Reply:
thanks, i missed that.